src/Microsoft.DotNet.Darc/src/DarcLib/Models/Darc/DependencyFlowGraph.cs

```diff
@@ -127,6 +128,36 @@ public void RemoveEdge(DependencyFlowEdge edge)
             }
         }
 
+        public async Task MarkToolingEdges(IRemoteFactory remoteFactory, ILogger logger)
+        {
+            var stopwatch = new Stopwatch();
+            stopwatch.Start();
+
+            foreach (var edge in Edges)
+            {
+                var lastAppliedBuildId = edge.Subscription.LastAppliedBuild?.Id;
+                if (lastAppliedBuildId.HasValue)
+                {
+                    var remote = await remoteFactory.GetRemoteAsync(edge.To.Repository, logger);
```

Comment:
This is concerning to me. Why aren't we using the barOnlyRemote? This requires things to connect to github/azdo/have to have a git instance (please correct me if I'm wrong about how the this works, but my understanding is that it is using a file manager to parse the versions details xml) when all of the information we want is just sitting in the bar database in the BuildDependencies table. We have had trouble in the past with using the version details xml files for remote calls like that and I want to understand if it is needed.

How I envisioned this working was that we would just query the BuildDependencies table. The BuildDependencies table has BuildId, DependentBuildId, and IsProduct: all that we need for determining if an edge is a tooling edge. So we'd have a function that takes the To.Repository, To.Branch and LastAppliedBuildId. We'd have a REST api function that takes that info and does the right query (ie something like

```
select BuildDependencies.IsProduct 
from BuildDependencies
join Builds on BuildDependencies.BuildId = Builds.Id
where BuildDependencies.DependentBuildId = '{LastAppliedBuildId}'
and Builds.GitHubRepository = '{To.Repository}' 
and Builds.GitHubBranch = '{To.Branch}'
```

but in Linq/C#), and takes the last time that dependent build id was applied. That would be the result for that edge.

Reply:
@adiaaida Yeah, it looks like a better idea. Just a few questions:

1. Can there be more that one row that matches this query? If yes then what to do then - take the latest build?
2. Would you suggest to create a new API endpoint for this query or add it into some existing one?

Reply:
After discussion with @adiaaida we decided to change it so that darc queries the graph from `/channels/{id}/graph` endpoint instead of building it separately. This should reduce duplication of code and make any changes in the algorithm easier because in API we have direct access to BAR database. This is already implemented in latest changes that I pushed.